txcache/crossTxCache.go

```diff
@@ -124,28 +127,12 @@ func (cache *CrossTxCache) GetTransactionsPoolForSender(_ string) []*WrappedTran
 	return make([]*WrappedTransaction, 0)
 }
 
-// RegisterEvictionHandler registers a handler which will be called when a tx is evicted from cache
-func (cache *CrossTxCache) RegisterEvictionHandler(handler func(hash []byte)) error {
-	if handler == nil {
-		return common.ErrNilEvictionHandler
-	}
-
-	cache.mutEvictionHandlers.Lock()
-	cache.evictionHandlers = append(cache.evictionHandlers, handler)
-	cache.mutEvictionHandlers.Unlock()
-
+// Close closes the eviction worker pool
+func (cache *CrossTxCache) Close() error {
+	cache.evictionWorkerPool.Stop()
```

Comment:
based on the lib documentation, `Stop()` will execute the current executing jobs and discard the ones in the queue. `StopWait()` will wait for all pending jobs to be executed. Not sure which one we should use

Reply:
used Stop() as I was thinking that it will be called when node will be stopped, thus no need to wait for the ones pending